### ISSUE TITLE
Color tint doesn't work on Android M

### ISSUE THREAD

**Issue by glodos** [negusoft/GreenMatter#5]
Hi, I just wanted to report that color tint does not longer work on Android M. It's caused by changes in the inflate() method of the ColorStateList class. I haven't found a way to workaround it yet, hope you will figure it out.

**Comment by blurkidi** [comment]
Indeed, I tried the lib and I was pretty disappointed that the hack to replace colors doesn't work on Android M :(. The problem, as you mention, is that Resources.getColor() is no longer called when loading a ColorStateList from XML. And I am afraid that I didn't find a way to fix that.

Hopefully, I think I might have an ace up your sleeve. I found another method to override colors at runtime. I still need to work it out but it might be a way to keep the functionality of modifying the theme colours at runtime.

In the worst case, this functionality will be dropped and the lib will become a Material backport. But that would be sad :/

**Comment by glodos** [comment]
Can you tell me what's the method name you've found? I'd like to have a deeper look at it.

**Comment by blurkidi** [comment]
The idea is to access the cache of ColorStateList using reflection and preload some colors. Such a way that the key is the color to be replaced and the value is the actual color. Like this when ColorStateList.valueOf() is called, the color is replaced. You can access the cache like this:

```
private SparseArray<WeakReference<ColorStateList>> initColorCache() {
    try {
        Field field = android.content.res.ColorStateList.class.getDeclaredField("sCache");
        field.setAccessible(true);
        Object arrayValue = field.get(null);

        return (SparseArray<WeakReference<ColorStateList>>)arrayValue;
    } catch (NoSuchFieldException e) {
        return null;
    } catch (IllegalAccessException e) {
        return null;
    }
}
```

The technique works, but it is quite limited. ColorStateList is not only initialised with valueOf(), so there are some cases were colors are not replaced properly.

**Comment by glodos** [comment]
Thanks, that gives me some hope. I'll give it a try after flashing the M again,

**Comment by fic212** [comment]
Hello guys,
Any update on this? This is great library and I realy hope that I would be able to use it on M.. 

Thx

**Comment by N-Olifer** [comment]
I am waiting for updates too. That great library is unique way to implement some color features in my app.

**Comment by glodos** [comment]
I'm sorry to disappoint you guys, but this library has no future. I had to manually tint each view, no other way will work.

**Comment by ThePreviousOne** [comment]
Not that I know how, but ASMDEX http://asm.ow2.org/asmdex-index.html could work
